### PR TITLE
Use formFields for storing formUrl query parameters

### PR DESCRIPTION
**Pull request by andrenpaes**

### CODE REVIEW

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenHostedPaymentPageFormDescriptor.java [hunk 1]
```diff
@@ -17,13 +17,19 @@
...
 
 public class AdyenHostedPaymentPageFormDescriptor extends PluginHostedPaymentPageFormDescriptor {
 
+    public AdyenHostedPaymentPageFormDescriptor(UUID kbAccountId, String formMethod, String formUrl, List<PluginProperty> formFields, List<PluginProperty> properties) {
+        super(kbAccountId, formMethod, formUrl, formFields, properties);
+    }
+
     public AdyenHostedPaymentPageFormDescriptor(final UUID kbAccountId, final String formUrl, final Map<String, String> queryParams) throws URISyntaxException {
         super(kbAccountId, formUrl, queryParams);
     }
```
Comment (sbrossie) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenHostedPaymentPageFormDescriptor.java]: Is that CTOR still used? If not, should we delete it?

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java [hunk 2]
```diff
@@ -500,12 +502,7 @@ public HostedPaymentPageFormDescriptor buildFormDescriptor(final UUID kbAccountI
...
 
         final String formUrl = hostedPaymentPagePort.getFormUrl(paymentData);
-
-        try {
-            return new AdyenHostedPaymentPageFormDescriptor(kbAccountId, formUrl, formParameter);
-        } catch (final URISyntaxException e) {
-            throw new PaymentPluginApiException("Unable to generate valid HPP url", e);
-        }
+        return new AdyenHostedPaymentPageFormDescriptor(kbAccountId, "GET", formUrl, PluginProperties.buildPluginProperties(formParameter), Collections.<PluginProperty>emptyList());
     }
```
Comment (sbrossie) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: We seem to already have [constants defined](https://github.com/killbill/killbill-plugin-framework-java/blob/master/src/main/java/org/killbill/billing/plugin/api/payment/PluginHostedPaymentPageFormDescriptor.java#L35) for `GET`.

**Comment by sbrossie** [comment]
Looks good. Just have some minor remarks. Also [travis seems to be failing](https://travis-ci.org/andrenpaes/killbill-adyen-plugin/builds/124289538)

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java (hunk 2 above)
Comment (andrenpaes) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Unfortunately those constants are protected. Maybe there's some other I can use?

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenHostedPaymentPageFormDescriptor.java (hunk 1 above)
Comment (andrenpaes) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenHostedPaymentPageFormDescriptor.java]: Good catch!

File: src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java (hunk 2 above)
Comment (sbrossie) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Ah bummer... not that important. Either submit PR to make them public or leave it the way it is.

Comment (andrenpaes) [inline src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java]: Created the PR: https://github.com/killbill/killbill-plugin-framework-java/pull/7

**Comment by andrenpaes** [comment]
@sbrossie Can you take another look?
I guess it's a workaround on the visibility of the constant, but I think it's actually a better solution compared to what I had done before.

Regarding travis, maybe there's some config I'm missing? I ran the failing tests in my machine using a valid adyen config, and they all passed.

**Pull request merged**